### DISCUSSION TITLE
Ignore all methods

### DISCUSSION THREAD

**Discussion by pabermod** [EWSoftware/SHFB#1067] [Ideas]
Hi,

In my project I only want to document public properties, not methods. It seems the only way to do this is by unchecking each method of all types in the API filter, but this is a manual action that is not scalable.

It would be good to have an option in the **Visibility** section to indicate whether you want to generate documentation for public methods or not.

**Comment by EWSoftware** [comment]
I don't really see much use for excluding by member type.  You can probably use the [XPath Reflection File Filter Plug-In](https://ewsoftware.github.io/SHFB/html/cd68ef02-3493-4af6-96de-1957b0aaf858.htm) to remove the unwanted elements.  If your public methods don't have XML comments or are tagged with the `exclude` XML comment element, the [Documented Entities Only Plug-In](https://ewsoftware.github.io/SHFB/html/1c05f8f0-71f3-420a-b546-a438c435b759.htm) would also work.

**Reply by pabermod** [reply]
Thank you, I will take a look at the first plugin
